src/__tests__/app/choose-a-template-type/page.test.tsx

```diff
@@ -39,3 +58,23 @@ test('ChooseTemplatePage', async () => {
 
   expect(container.asFragment()).toMatchSnapshot();
 });
+
+test('ChooseATemplateTypePage - should handle invalid session', async () => {
+  mockSessionSupplier.mockSession = undefined;
+  const redirectSpy = jest.spyOn(nav, 'redirect');
+
+  let caughtError: Error | undefined;
+  try {
+    await ChooseATemplateTypePage({
+      params: {
+        sessionId: 'session-id',
+      },
+    });
+  } catch (error) {
+    caughtError = error as Error;
+  }
+
+  expect(caughtError).toBeTruthy();
+  expect(caughtError!.message).toBe('Simulated redirect');
```

Comment:
i think lines 66-77 could all just be:

```
await expect(ChooseATemplateTypePage({
      params: {
        sessionId: 'session-id',
      },
    })).rejects.toThrow('Simulated redirect');
```

Reply:
same applies everywhere this reassigning error pattern happens

Reply:
this still appears in the CreateNhsAppTemplate tests and appears as a comment on the ChooseATemplateType tests